### DISCUSSION TITLE
Disable copilot per workspace

### DISCUSSION THREAD

**Discussion by tekumara** [community/community#47991] [Copilot Conversations] [answered]
### Select Topic Area

Product Feedback

### Body

I'd like to disable copilot per workspace, so I can use it for open-source projects but not private/work projects.

**Comment by AsafMeizner** [comment]
in vs code there is a GitHub icon in bottom right where you can quickly turn it on and off,

hope this helps :)

**Reply by tekumara** [reply]
When I click on that icon in the bottom right it updates `github.copilot.enable` in my user settings for **all** workspaces though. 
I was hoping for something that allowed me to disable it only in the current workspace.

**Reply by AsafMeizner** [reply]
That's the closest you can get to that

**Reply by jarrodkohl** [reply]
I believe if you go into settings. And disable CoPilot, (restart VScode) then when you go to "enable" there is a little dropdown and you can select Enable or Enable(workspace) which may be what you are looking for.

**Reply by tekumara** [reply]
This is the popup I get when clicking the Copilot icon in the status bar on the bottom:

<img width="445" alt="CleanShot 2023-02-27 at 11 35 17@2x" src="https://user-images.githubusercontent.com/125105/221447468-4bc3c066-9d5f-4d59-9789-aa4dc3c87e0a.png">

Where do I find the dropdown?

**Reply by jgcaruso** [reply, marked as answer]
@tekumara That setting is for disabling Copilot on specific languages within the extension.

I believe the suggestion was to disable the extension itself, and only enable the extension for specific workspaces. This way the extension only runs for the workspaces you enable it for, and then "globally" enabling Copilot in that popup in the bottom right will just apply to those workspaces.

<img width="623" alt="image" src="https://user-images.githubusercontent.com/4081020/228581444-d297b180-5b78-4b8b-9009-dd6b297ccc14.png">

I'm trying this out myself, and it appears to work as desired. I don't get the Copilot icon in the bottom right of the window for any workspaces except those I've enabled the plugin on.

**Reply by tekumara** [reply]
Ah thank you for showing me this! I can confirm that works.

Any idea where on disk this setting is stored? I'd like to automate this.

**Reply by jgcaruso** [reply]
> Any idea where on disk this setting is stored? I'd like to automate this.

I had the same question, I didn't notice it in the workspace file. Didn't dig much further though. If you find out please share 😄

**Reply by orestiskapar** [reply]
Any way to control this from settings.json?

**Reply by tekumara** [reply]
Looks like there's an open feature request to enable/disable extensions by workspace see https://github.com/microsoft/vscode/issues/40239

**Reply by jsunico** [reply]
Has this been changed recently? I can't use the "Enable (Workspace)" feature.

![image](https://github.com/community/community/assets/65151494/1b505922-63b1-4289-9664-b3797db17e9e)

**Reply by jgcaruso** [reply]
> Has this been changed recently? I can't use the "Enable (Workspace)" feature.
> 

I think you first need to have a workspace file created. If you just started a new window and didn't load a workspace that option seems to be disabled. `File -> Save Workspace As...` should create a workspace file for you, then you can start disabling extensions.

**Reply by eevmanu** [reply]
Thanks for the answer, this is what I was looking for: 
### disable copilot globally

click on `Disable`

### enable copilot per workspace

click on `Enable (workspace)`

### disable for all file types at user settings level

despite having copilot enabled at workspace level

go to `~/.config/Code/User/settings.json`

```json
{
    // ...
    "github.copilot.enable": {
      "*": false,
    },
    // ...
}
```

### enable copilot per specific file types at workspace settings level

to manage copilot config at workspace settings level and not at user settings level

go to folder settings file - `.vscode/settings.json`

```json
{
    // ...
    "github.copilot.enable": {
        "*": false,
        "python": true,
        "markdown": true,
        "javascript": true,
        "shellscript": true,
    },
    // ...
}
```

or multi-root workspace settings file - `.code-workspace` file

```json
{
    // ...
    "settings": {
        // ...
        "github.copilot.enable": {
            "*": false,
            "python": true,
            "markdown": true,
            "javascript": true,
            "shellscript": true,
        },
        // ...
    },
    // ...
}
```

edit: update 2025-03-19 - not able to setup `github.copilot.enable` in workspace settings anymore since:

> This setting has an application scope and can be set only in the user settings file.

```sh
$ jq '.contributes.configuration[0].properties["github.copilot.enable"].scope' \
  "${HOME}/.vscode/extensions/github.copilot-1.288.0/package.json"
"application"
```

so the only way to set this is in user settings 🤷‍♂️

**Comment by shahul01** [comment]
To disable it per workspace
create a folder called `.vscode`, if not already exist
create a file called settings.json, if not already exist

That json file should have this line - `github.copilot.editor.enableAutoCompletions` set to `false`
like so,
```json
{
    // other settings
    "github.copilot.editor.enableAutoCompletions": false,
    // other settings
}
```
